bycycle/plts/cyclepoints.py

```diff
@@ -0,0 +1,187 @@
+    xlim : tuple of (float, float), optional, default: None
+        Start and stop times.
+    plot_sig : bool, optional, default: True
+        Plots the raw signal.
+    plot_extrema :  bool, optional, default: True
+        Plots peaks and troughs.
+    plot_zerox :  bool, optional, default: True
+        Plots zero-crossings.
+    ax : matplotlib.Axes, optional, default: None
+        Figure axes upon which to plot.
+    **kwargs
+        Keyword arguments to pass into `plot_time_series`.
+
+    Notes
+    -----
+    Default keyword arguments include:
+
+    - ``figsize``: tuple of (float, float), default: (15, 3)
+    - ``xlabel``: str, default: 'Time (s)'
+    - ``ylabel``: str, default: 'Voltage (uV)
+
+    """
+
+    # Set default kwargs
+    figsize = kwargs.pop('figsize', (15, 3))
+    xlabel = kwargs.pop('xlabel', 'Time (s)')
+    ylabel = kwargs.pop('ylabel', 'Voltage (uV)')
+
+    # Set times and limits
+    times = np.arange(0, len(sig) / fs, 1 / fs)
+    xlim = (times[0], times[-1]) if xlim is None else xlim
+
+    # Determine extrema/zero-crossing times and signals
+    center_e, side_e = get_extrema(df)
+
+    df = limit_df(df, fs, xlim)
+    sig, times = limit_sig_times(sig, times, xlim)
+
+    # Extend plotting based on given arguments
+    x_values = []
+    y_values = []
+    colors = ['k']
+
+    if plot_extrema:
+
+        ps = df['sample_' + center_e].values
+        ts = df['sample_last_' + side_e].values
+        ts = np.append(ts, df['sample_next_' + side_e].values[-1])
+
+        # Cycles are kept if any cyclepoint is within tlims, this ensures all points to be plotted
+        #   are within the x limits.
+        ps = ps[(ps >= 0) & (ps < (xlim[1] - xlim[0]) * fs)]
+        ts = ts[(ts >= 0) & (ts < (xlim[1] - xlim[0]) * fs)]
+
+        x_values.extend([times[ps], times[ts]])
+        y_values.extend([sig[ps], sig[ts]])
+        colors.extend(['b', 'r'])
+
+    if plot_zerox:
+        zerox_rise = df['sample_zerox_rise'].values
+        zerox_rise = zerox_rise[(zerox_rise >= 0) & (zerox_rise < (xlim[1] - xlim[0]) * fs)]
+        zerox_decay = df['sample_zerox_decay'].values
+        zerox_decay = zerox_decay[(zerox_decay >= 0) & (zerox_decay < (xlim[1] - xlim[0]) * fs)]
+
+        x_values.extend([times[zerox_rise], times[zerox_decay]])
+        y_values.extend([sig[zerox_rise], sig[zerox_decay]])
+        colors.extend(['g', 'm'])
+
+    # Allow custom colors to overwrite default
+    colors = kwargs.pop('colors', colors)
+
+    # Plot cycle points
+    if ax is None:
+        fig, ax = plt.subplots(figsize=figsize)
+
+    if plot_sig:
+        plot_time_series(times, sig, colors=colors[0], ax=ax)
+
+    colors = colors[1:] if plot_sig is True else colors
+
+    plot_time_series(x_values, y_values, ax=ax, colors=colors, xlabel=xlabel,
+                     ylabel=ylabel, marker='o', ls='', **kwargs)
+
+
+@savefig
+def plot_cyclepoints_array(sig, fs, xlim=None, ps=None, ts=None, zerox_rise=None,
```

Comment:
The API labels of `ps`, `ts`, `zerox_rise`, `zerox_decay` feels a little inconsistent to me. In other functions we've updated from `p` or `ps` to more explicitly say `peaks`, and since we don't say `extrema_peaks`, I'm not sure we need to note `zerox`. 

What about just calling them `peaks`, `troughs`, `rises`, `decays`, or something like that? To me, that feels the most consistent, and straight-forward. 

Also, I think I would put the `xlim` arg after all the cyclepoint ones.